Sources/MapboxMaps/Foundation/Camera/CameraManager+CameraAnimatorDelegate.swift

```diff
@@ -15,12 +15,14 @@ extension CameraManager: CameraAnimatorDelegate {
     ///   - timingParameters: The object providing the timing information. This object must adopt the `UITimingCurveProvider` protocol.
     ///   - animationOwner: Property that conforms to `AnimationOwnerProtocol` to represent who owns that animation.
     /// - Returns: A class that represents an animator with the provided configuration.
-    public func makeCameraAnimator(duration: TimeInterval,
-                                   timingParameters parameters: UITimingCurveProvider,
-                                   animationOwner: AnimationOwner = .unspecified) -> CameraAnimator {
+    public func makeAnimator(duration: TimeInterval,
+                             timingParameters parameters: UITimingCurveProvider,
+                             animationOwner: AnimationOwner = .unspecified,
+                             animations: @escaping (inout CameraTransition) -> Void) -> BasicCameraAnimator {
         let propertyAnimator = UIViewPropertyAnimator(duration: duration, timingParameters: parameters)
-        let cameraAnimator = CameraAnimator(delegate: self, propertyAnimator: propertyAnimator, owner: animationOwner)
-        cameraAnimators.add(cameraAnimator)
+        let cameraAnimator = BasicCameraAnimator(delegate: self, propertyAnimator: propertyAnimator, owner: animationOwner)
+        cameraAnimator.addAnimations(animations)
+        mapView?.cameraAnimatorsHashTable.add(cameraAnimator)
```

Comment:
The fact that CameraManager is directly modifying the contents of MapView's `cameraAnimatorsHashTable` makes me nervous. I would recommend that we adopt a general guideline that any operation that impacts the mutable state of another class should flow through one of that class's interfaces, not the interface of the mutable state itself. This will make a big difference in keeping our code decoupled and easier to maintain. This is also one of the big drivers of all my feedback in this PR about making things private when possible.

Reply:
I agree with your comment. Let me think about it and come up with a solution.